### PR TITLE
🎉Destination-s3: added support for AWS Glue crawler

### LINKED ISSUE #4654: Destination S3: Allow special chars for table_prefix
Hi,
I am trying to write to S3 destination with keys formatted as `static_prefix/source=<namespace>/object=<table_name>/<filename>`, like this:
screenshot

aitbyte is normalizing the path to only contain letters and unserscores, resulting in paths like this:
```static_prefix/source_<namespace>/object_<table_name>/<filename>```

configured paths should not be modified

### PR DESCRIPTION
**Pull request by etsybaev**
## What
A user wants to use the [AWS Glue crawler](https://docs.aws.amazon.com/glue/latest/dg/crawler-s3-folder-table-partition.html) feature, but currently, all special symbols transform to "_" symbol.

## How
Updated logic to do not transform special symbols in stream name if user adds the "=" symbol in prefix.

Tested locally:
Selection_041

Selection_042

## 🚨 User Impact 🚨
No breaking changes expected

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by VitaliiMaltsev — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]
LGTM. please reformat the code and update the related documentation

File: airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java [hunk 1]
```diff
@@ -29,7 +29,13 @@ public static String getOutputPrefix(final String bucketPath, final String names
     if (namespace != null) {
       paths.add(NAME_TRANSFORMER.convertStreamName(namespace));
     }
-    paths.add(NAME_TRANSFORMER.convertStreamName(streamName));
+    if (streamName.contains("=")){
+      // case AWS Glue crawler
+      // more details https://docs.aws.amazon.com/glue/latest/dg/crawler-s3-folder-table-partition.html
+      paths.add(streamName);
+    }else {
+      paths.add(NAME_TRANSFORMER.convertStreamName(streamName));
```
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java:37]: Shouldn't we create a `NamingConventionTransformer nameTransformer` when instantiating the `S3Destination`?

Here are some examples of how it's done in other connectors:
- Postgres: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java#L38
- Snowflake: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java#L33
- MongoDB: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java#L64
- etc

Or at least, when we construct the consumer, we pass it to the factory:
- Redshift: https://github.com/airbytehq/airbyte/blob/ea92a24da35ff9a32288bd96fc00a82ff7cad481/airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftCopyS3Destination.java#L59

Comment (girarda) [reply airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java:37]: I agree. We should hide that complexity in a custom NameTransformer

**Review by girarda — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java (hunk 1 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java:37]: As far as I understood, this converter is used for creating a stream name only.  A user wants to use some specific feature([AWS Glue crawler). That's why I removed any auto-replacement if "=" symbol is used

Comment (girarda) [reply airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3OutputPathHelper.java:37]: @etsybaev the logic is sound, but we should encapsulate it into a `S3NameTransformer`

File: airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3NameTransformer.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  public static final String NON_ALPHANUMERIC_AND_UNDERSCORE_DASH_EQUAL_PATTERN = "[^\\p{Alnum}_=-]";
+
+  /**
+   * Converts any UTF8 string to a string with only alphanumeric and "=", "-" and "_" characters without preserving
+   * accent characters.
+   *
+   * @param input string to convert
+   * @return cleaned string
+   */
+  public String convertStreamName(final String input) {
```
Comment (girarda) [inline airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3NameTransformer.java:31]: @etsybaev can we add a couple of unit tests for the regex?

**Review by girarda — APPROVED** [review]

**Pull request merged**